src/java/org/apache/cassandra/cql3/Vectors.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.cassandra.db.marshal.VectorType;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+public class Vectors
+{
+    private Vectors() {}
+
+    private static AbstractType<?> elementsType(AbstractType<?> type)
+    {
+        return ((VectorType<?>) type.unwrap()).getElementsType();
+    }
+
+    private static ColumnSpecification valueSpecOf(ColumnSpecification column)
+    {
+        return new ColumnSpecification(column.ksName, column.cfName, new ColumnIdentifier("value(" + column.name + ")", true), elementsType(column.type));
+    }
+
+    public static class Literal extends Term.Raw
+    {
+        private final List<Term.Raw> elements;
+
+        public Literal(List<Term.Raw> elements)
+        {
+            this.elements = elements;
+        }
+
+        @Override
+        public TestResult testAssignment(String keyspace, ColumnSpecification receiver)
+        {
+            if (!(receiver.type instanceof VectorType))
+                return AssignmentTestable.TestResult.NOT_ASSIGNABLE;
+            VectorType<?> type = (VectorType<?>) receiver.type;
+            if (elements.size() != type.dimension)
+                return AssignmentTestable.TestResult.NOT_ASSIGNABLE;
+            ColumnSpecification valueSpec = valueSpecOf(receiver);
+            return AssignmentTestable.TestResult.testAll(receiver.ksName, valueSpec, elements);
+        }
+
+        @Override
+        public Term prepare(String keyspace, ColumnSpecification receiver) throws InvalidRequestException
+        {
+            if (!(receiver.type instanceof VectorType))
+                throw new InvalidRequestException(String.format("Invalid vector literal for %s of type %s", receiver.name, receiver.type.asCQL3Type()));
+            VectorType<?> type = (VectorType<?>) receiver.type;
+            if (elements.size() != type.dimension)
+                throw new InvalidRequestException(String.format("Invalid vector literal for %s of type %s; expected %d elements, but given %d", receiver.name, receiver.type.asCQL3Type(), type.dimension, elements.size()));
+
+            ColumnSpecification valueSpec = valueSpecOf(receiver);
+            List<Term> values = new ArrayList<>(elements.size());
+            boolean allTerminal = true;
+            for (Term.Raw rt : elements)
+            {
+                if (!rt.testAssignment(keyspace, valueSpec).isAssignable())
+                    throw new InvalidRequestException(String.format("Invalid vector literal for %s: value %s is not of type %s", receiver.name, rt, valueSpec.type.asCQL3Type()));
+
+                Term t = rt.prepare(keyspace, valueSpec);
+
+                if (t instanceof Term.NonTerminal)
+                    allTerminal = false;
+
+                values.add(t);
+            }
+            DelayedValue value = new DelayedValue(type, values);
```

Comment:
nit: `DelayedValue` should be parameterized -> `DelayedValue<?> value = new DelayedValue<>(type, values);`